### ISSUE TITLE
m_iCashSpentThisRound is always nil

### ISSUE THREAD

**Issue by xavier-rodet** [markus-wa/demoinfocs-golang#126]
**Describe the bug**
Calling Player.CashSpentThisRound() is throwing error because m_iCashSpentThisRound is always nil

**To Reproduce**
Any demo

Code:
```go
func analyze(filePath string) {

    var isGameLive bool

    f, err := os.Open(filePath)
    if err != nil {
        panic(err)
    }
    defer f.Close()

    p := dem.NewParser(f)

    header, err := p.ParseHeader()
    if err != nil {
        panic(err)
    }

    mapName := header.MapName
    fmt.Printf("%s\n", mapName)

    // Match Started
    p.RegisterEventHandler(func(e events.MatchStart) {

        isGameLive = true
    })


    // Round Ended (when winner is decided, kills can still happens after it)
    p.RegisterEventHandler(func(e events.RoundEnd) {

        if (isGameLive) {

            // Iterate round's winner team
            if(e.WinnerState != nil) {

                winnerTeam := e.WinnerState.Members()
                for _, winnerPlayer := range winnerTeam {

                    fmt.Printf("Entity : %v\n", winnerPlayer.Entity.FindPropertyI("m_iCashSpentThisRound"))
                    fmt.Printf("CashSpent : %v\n", winnerPlayer.CashSpentThisRound())
                }
            }

            // Iterate round's loser team
            if(e.LoserState != nil) {

                loserTeam := e.LoserState.Members()
                for _, loserPlayer := range loserTeam {

                    fmt.Printf("Entity : %v\n", loserPlayer.Entity.FindPropertyI("m_iCashSpentThisRound"))
                    fmt.Printf("CashSpent : %v\n", loserPlayer.CashSpentThisRound())
                }
            }
        }
    })

    // Parse to end
    err = p.ParseToEnd()
}
```

**Expected behavior**
I should get the money spent by each players for each rounds

**Library version**
Current master

**Additional context**
Here's my output :

> de_inferno
> Entity : \<nil\>
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x80401e]
> 
> goroutine 24 [running]:
> github.com/markus-wa/demoinfocs-golang/common.(*Player).CashSpentThisRound(0xc0029e5600, 0xc0000a8008)
> 	/go/src/github.com/markus-wa/demoinfocs-golang/common/player.go:163 +0x6e
> main.analyze.func2(0xc0017a7560, 0x14, 0x308, 0xc00010d910, 0xc00010d8c8)
> 	/app/src/analyzer.go:650 +0x33e
> reflect.Value.call(0xc3f920, 0xc0001a91f0, 0x13, 0xd71635, 0x4, 0xc0000c5ab0, 0x1, 0x1, 0x1, 0x1545a00, ...)
> 	/usr/local/go/src/reflect/value.go:447 +0x461
> reflect.Value.Call(0xc3f920, 0xc0001a91f0, 0x13, 0xc0000c5ab0, 0x1, 0x1, 0x28, 0xc0059626c0, 0xc0000c5a68)
> 	/usr/local/go/src/reflect/value.go:308 +0xa4
> github.com/markus-wa/godispatch.(*Dispatcher).Dispatch(0xc000200090, 0xcf7a60, 0xc0059626c0)
> 	/go/src/github.com/markus-wa/godispatch/dispatch.go:48 +0x253
> github.com/markus-wa/demoinfocs-golang.gameEventHandler.dispatch(...)
> 	/go/src/github.com/markus-wa/demoinfocs-golang/game_events.go:47
> github.com/markus-wa/demoinfocs-golang.gameEventHandler.roundEnd(0xc000200000, 0x0, 0xc0001c4480, 0xc000c9eb40)
> 	/go/src/github.com/markus-wa/demoinfocs-golang/game_events.go:169 +0x1fc
> github.com/markus-wa/demoinfocs-golang.(*Parser).handleGameEvent(0xc000200000, 0xc000c9eb40)
> 	/go/src/github.com/markus-wa/demoinfocs-golang/game_events.go:31 +0xb1
> reflect.Value.call(0xc3ee20, 0xc0001a9180, 0x13, 0xd71635, 0x4, 0xc0000c5f68, 0x1, 0x1, 0x4, 0x1545a00, ...)
> 	/usr/local/go/src/reflect/value.go:447 +0x461
> reflect.Value.Call(0xc3ee20, 0xc0001a9180, 0x13, 0xc0000c5f68, 0x1, 0x1, 0x0, 0x0, 0x0)
> 	/usr/local/go/src/reflect/value.go:308 +0xa4
> github.com/markus-wa/godispatch.(*Dispatcher).Dispatch(0xc000200020, 0xd34920, 0xc000c9eb40)
> 	/go/src/github.com/markus-wa/godispatch/dispatch.go:48 +0x253
> github.com/markus-wa/godispatch.(*Dispatcher).dispatchQueue(0xc000200020, 0xc0000ab080)
> 	/go/src/github.com/markus-wa/godispatch/dispatch.go:97 +0xf5
> created by github.com/markus-wa/godispatch.(*Dispatcher).AddQueues
> 	/go/src/github.com/markus-wa/godispatch/dispatch.go:83 +0x151
> exit status 2

**Comment by markus-wa** [comment]
Yikes, I will try to look into it over the weekend. Thanks for the info.

**Comment by markus-wa** [comment]
Should be fixed in [v1.2.1](https://github.com/markus-wa/demoinfocs-golang/releases/tag/v1.2.1) now.

**Comment by xavier-rodet** [comment]
Awesome 👍
